### PR TITLE
Refactor raisepriority and track automatic prio bumps via history element

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

**Comment by bgeuken** [comment]
@openSUSE/open-build-service Please review

@adrianschroeter Speak now or be silent forever;-)

File: src/api/app/models/bs_request.rb [hunk 1]
```diff
@@ -1174,12 +1163,39 @@ def update_cache
...
     bs_request_actions.each do |action|
       action.check_action_permission!(opts[:skip_source])
       action.check_for_expand_errors!(!@addrevision.nil?)
       raisepriority(action.minimum_priority)
     end
+
+    return unless persisted?
+
+    # rubocop:disable Style/GuardClause
+    if priority_changed?
```
Comment (DavidKang) [inline src/api/app/models/bs_request.rb]: Why you disable rubocop here?

Comment (bgeuken) [reply src/api/app/models/bs_request.rb]: Because I am not going to write a guard claus for a 7 line statement :rofl:

**Comment by bgeuken** [comment]
Already spoke with Adrian today and he is fine with the approach => merging

**Comment by bgeuken** [comment]
[(2) P3: Add logging of request prio change via a history element](https://trello.com/c/UJJSGTnw/589-2-p3-add-logging-of-request-prio-change-via-a-history-element)

**Pull request merged**